smartsim/settings/base.py

```diff
@@ -367,12 +367,12 @@ def update_env(self, env_vars):
         :type env_vars: dict[str, Union[str, int, float, bool]]
         :raises TypeError: if env_vars values cannot be coerced to strings
         """
-        val_types = (str, int, float, bool)
+        val_types_union = (str, int, float, bool)
```

Comment:
The code in this function was updated on `develop`, but it seems that by fixing the conflicts, we reverted to an older version. I'd suggest to checkout the `develop` version, as it is more readable.